src/ncdiff/netconf.py

```diff
@@ -200,8 +207,112 @@ def sub(self):
             self.get_config_replace(ele1, ele2)
         else:
             self.node_sub(ele1, ele2, depth=0)
+        # add attribute at depth if diff_type is 'minimum-replace'
+        if self.diff_type == 'minimum-replace' and self.replace_xpath:
+            namespaces = self.device._get_ns(ele1)
+            logger.debug("Namespaces:\n{}".format(json.dumps(namespaces, indent=2)))
+            self.add_attribute_by_xpath(ele1, self.replace_xpath, 'operation', 'replace', namespaces)
+        elif self.diff_type == 'minimum-replace':
+            self.add_attribute_at_depth(ele1, self.replace_depth+1, 'operation', 'replace')
         return ele1
 
+    def add_attribute_at_depth(self, root, depth, attribute, value):
+        '''add_attribute_at_depth
+        High-level api: Add an attribute to all nodes at a specified depth.
+        Parameters
+        ----------
+        root : `Element`
+            The root of a config tree.
+        depth : `int`
+            The depth of nodes to be added with an attribute.
+        attribute : `str`
+            The name of the attribute to be added.
+        value : `str`
+            The value of the attribute to be added.
+        Returns
+        -------
+        None
+        '''
+        # Initialize current_depth as -1
+        current_depth = -1
+
+        # Create a list with the root node and its depth (0)
+        nodes_to_visit = [(root, current_depth + 1)]
+
+        # While there are nodes to visit
+        while nodes_to_visit:
+            # Pop the first node and its depth from the list
+            node, current_depth = nodes_to_visit.pop(0)
```

Comment:
`nodes_to_visit` was defined above with one element (line 240). This while loop should only loop once. Do I miss something here?

Reply:
thank you for your comment. within the loop, we potentially add more nodes to nodes_to_visit (for example, when we encounter a node that has children). This means the loop can iterate more than once depending on the structure of the XML document we're processing. If the XML only has one level, the loop will indeed only iterate once. But for nested XML structures, the loop allows us to traverse and process all nodes.